### DISCUSSION TITLE
Remaining asok files of rgw will effect  the ceph exporter?

### DISCUSSION THREAD

**Discussion by zhucan** [rook/rook#12072] [Q&A]
1. Remaining asok files of rgw under forced or abnormal exit.
```
[root@node01 ceph]# ls /run/ceph/ceph-client.rgw.os.dsglczutvqsgowpz.a.
ceph-client.rgw.os.dsglczutvqsgowpz.a.13.94703695074760.asok    ceph-client.rgw.os.dsglczutvqsgowpz.a.657.94586861336008.asok
ceph-client.rgw.os.dsglczutvqsgowpz.a.2172.94328535029192.asok  
```
the valid asock file is: ceph-client.rgw.os.dsglczutvqsgowpz.a.2172.94328535029192.asok  

`ceph      2172     0 23 Apr08 ?        19:20:09 radosgw --fsid=f7f1c8ba-f793-436b-bb73-0964108a30c1 --keyring=/etc/ceph/keyring-store/keyring --log-to-stderr=true --err-to-stderr=true --mon-cluster-log-to-stderr=true --log-stderr-prefix=debug  --default-log-to-file=false --default-mon-cluster-log-to-file=false --mon-host=[v2:x.x.x.x:3300,v1:x.x.x.x:6789],[v2:x.x.x.x:3300,v1:x.x.x.x:6789],[v2:x.x.x.x:3300,v1:x.x.x.x:6789] --mon-initial-members=c,a,b --id=rgw.os.dsglczutvqsgowpz.a --setuser=ceph --setgroup=ceph --foreground --rgw-frontends=beast port=80 ssl_port=9000 ssl_certificate=/etc/ceph/private/rgw-cert.pem ssl_private_key=/etc/ceph/private/rgw-key.pem --host=rook-ceph-rgw-os-dsglczutvqsgowpz-a-64cd5fd9b6-l7vz8 --rgw-mime-types-file=/etc/ceph/rgw/mime.types --rgw-realm=os-dsglczutvqsgowpz --rgw-zonegroup=os-dsglczutvqsgowpz --rgw-zone=os-dsglczutvqsgowpz`

2. And  the ceph-export use the asok under the dir "/run/ceph" to get metrics, like "ceph-exporter --sock-dir /run/ceph --port 9926 --prio-limit 5 --stats-period 5"
3. but some sock under the dir is invalid, is there any bad effects?

**Comment by travisn** [comment]
@avanthakkar This sounds related to the new exporter and the daemon socket on each node for the metrics collection.
I don't expect these leftover files to cause any issues, but would be good to have them cleaned up.
